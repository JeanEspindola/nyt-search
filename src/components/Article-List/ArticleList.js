/*
* Article List Presentation Component
*
* Renders the list of returned actions, performs page navigation on results.
*
* */
import React from 'react';
import { ListGroup, ListGroupItem } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import PageNavigation from '../Page-Navigation/PageNavigationContainer';
import LoaderHOC from '../HOC/LoadingHOC';

const ArticleList = ({ list }) => (
  <div>
    <h4>Results:</h4>
    <ListGroup>
      {
        list.map(item => (
          <ListGroupItem
            key={item.id}
          >
            <Link to={{
              pathname: '/detail',
              article: item,
            }}
            >
              {item.headline.main}
            </Link>
          </ListGroupItem>
        ))
      }
    </ListGroup>
    <PageNavigation />
  </div>
);

ArticleList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default LoaderHOC('list', 'loading')(ArticleList);
