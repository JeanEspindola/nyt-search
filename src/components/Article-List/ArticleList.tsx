import React from 'react'
import { Link } from 'react-router-dom'
import PageNavigation from '../Page-Navigation/PageNavigation'
import Loading from '../../components/Screens/Loading'
import { useSelector } from 'react-redux'
import { RootStateType } from '../../redux/rootTypes'
import { FormattedMessage } from 'react-intl'
import styles from './ArticleList.module.scss'

const ArticleList = () => {
  const { list, loading } = useSelector((state: RootStateType) => state.articleList)
  let component: React.ReactNode

  if (loading) {
    component = <Loading />
  } else if (list === null || list === undefined || list.length === 0) {
    component = (
      <p className={styles.noResults}>
        <FormattedMessage id={'noResults'} />
      </p>
    )
  } else {
    component = (
      <>
        <div className={styles.title}>
          <h4>
            <FormattedMessage id={'results'} />
          </h4>
        </div>
        {list.map(item => (
          <div key={item.id} className={styles.listItem}>
            <Link to={{ pathname: '/detail', state: { article: item } }}>{item.headline.main}</Link>
          </div>
        ))}
        <PageNavigation />
      </>
    )
  }

  return <React.Fragment>{component}</React.Fragment>
}

export default ArticleList
