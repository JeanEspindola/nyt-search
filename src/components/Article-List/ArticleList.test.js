import React from 'react';
import { shallow } from 'enzyme';
import ArticleList from './ArticleList';

describe('ArticleList', () => {
  const mockOnQuerySearch = jest.fn;
  const page = 2;
  const query = 'search';
  const list = [];

  let Component;

  beforeEach(() => {
    Component = shallow(<ArticleList
      list={list}
      page={page}
      query={query}
      onQuerySearch={mockOnQuerySearch}
    />);
  });

  it('renders ArticleList without crashing', () => {
    expect(Component.length).toBeTruthy();
  });
});
