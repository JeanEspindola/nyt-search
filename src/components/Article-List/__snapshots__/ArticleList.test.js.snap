// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders ArticleList without crashing 1`] = `
<ArticleList
  list={
    Array [
      Object {
        "headline": Object {
          "main": "a",
        },
        "id": 1,
      },
      Object {
        "headline": Object {
          "main": "a",
        },
        "id": 2,
      },
    ]
  }
/>
`;
